### ISSUE TITLE
[6.x] Global increased damage to lower level creatures

### ISSUE THREAD

**Issue by StormBytePP** [TrinityCore/TrinityCore#15244]
After Cataclysm, instead of raising exponentially the stat numbers, Blizzard decided to implement a global damage increase system, that "nerf" enabled you to do more damage to lower level creatures, instead of requiring you a very high number of stats.

This also, enabled you to solo many dungeons/raids you could not before (like ICC, for instance), and also, enables you to kill some 50 level mobs in 1 hit.
However, in TC this seems not well implemented, or at least not in all spells.

I created a sniff on a level 100 shaman, COMPLETELLY UNEQUIPED/NAKED, so the equipment does not interfere in stats, just basic stats of the lvl 100 draenei shaman. (Sniff is: http://community.trinitycore.org/topic/11731-tanaris-low-level-creature-kill-test/ )

This is just one example, but I gathered the data also in paper (the numbers).
Also, in retail since I did 1 hit kill, multistrike did not apply, so even better for calculations?

These are the numbers: (if I don't mention any other spell, means I did not have any other buffs, nor I applied any other spell before)

Zone: Tanaris
- Target: Sand Lasher, lvl 47, ID: 44569
  - Spell: Lava Burst ID: 5105
    - RETAIL
      - 1950 fire (77263 overkill) ( critical )
      - 1 hit kill
    - TC
      - 1330 fire
      - 620 fire (710 overkill)
      - 2 hit kill
  - Spell: Lightning bolt ID: 403 (of course another mob, as previous is dead)
    - RETAIL
      - 1950 nature (115775 overkill) (critical)
      - 1 hit kill
    - TC
      - 1950 nature (27 overkill)
      - 1 hit kill

As you can see here, the numbers are **VERY** different, and even, I need 2 hit kills in TC.
I am not sure about which formula to apply for the increased damage system, but it is fundamentally broken, and can be easily noted/reproduced, it is not just a slight bug, but a very big difference one.

Tested on rev 8ad51b5a6eab5400723b58dd3ce9986e4f824bb1

**Comment by Zerixx** [comment]
Yeah, I've been looking at this, actually. Celestalon (US CM) said this in a tweet. The formulas for the scaling are as such:

To determine how much damage you take from a monster you use this formula. Basically, the damage you take is based on the level difference between you and the creature. If the creature is not declared WoD in the database then the function runs as such. If the monster is WoD then the damage taken factor remains at 1.0.

```
LevelDiff = PlayerLevel - CreatureLevel;
if (CreatureExpansion < EXPANSION_WARLORDS_OF_DRAENOR)
    DamageTakenFactor = std::max(1.0f - 0.1f * LevelDiff, 0.1f); // 10% DR per level diff, with a floor of 10%
else
    DamageTakenFactor = 1.0f;
```

To determine how much damage you deal to the monsters you use this. Basically, when you deal damage to monsters NOT WoD, the level difference determines your damage you deal to the monster. E.g., the level difference is < 5 then the damage ranges from 1.0625x to 1.25x normal damage, and so on.

```
LevelDiff = PlayerLevel - CreatureLevel;
if (CreatureExpansion >= EXPANSION_WARLORDS_OF_DRAENOR) then
    DamageDealtFactor = 1.0f
elseif (LevelDiff < 5)
    DamageDealtFactor = 1.0f + 0.0625f * LevelDiff; // Ranges from 1.0625 to 1.25 vs. 1-4 LevelDiffs
elseif (LevelDiff < 10) then
    DamageDealtFactor = 1.5f + 0.5f * LevelDiff; // Ranges from 4.0 to 6.0 vs. 5-9 LevelDiffs
else
    DamageDealtFactor = 16.5f; // Maximum factor of 16.5 vs. 10+ LevelDiffs
}
```

"These two formulations aren't the only ones that will be applied, however. Celestalon mentioned that there was a desire to allow for power increase during an expansion before Mists of Pandaria to still provide benefit while dealing with creatures that were outleveled, and so the following formula also exists and is used when it would be more of a benefit to a player."
Note: I haven't even looked at this one yet.

```
MaxPlayerLevelsByExpansion = {69, 79, 84, 89, 0, 0};
IntendedItemLevelByExpansion = {65, 115, 200, 346, 0, 0};
MaxPlayerLevel = MaxPlayerLevelsByExpansion[proto->expansion]; // for the creature expansion
IntendedItemLevel = IntendedItemLevelByExpansion[proto->expansion]; // ^^^

if (PlayerLevel <= MaxPlayerLevel && PlayerEquippedItemLevel > IntendedItemLevel)
{
    AlternateDamageTakenFactor = 1 - 0.01 * (PlayerEquippedItemLevel - IntendedItemLevel);
    AlternateDamageDealtFactor = 1 + 5/3*0.01 * (PlayerEquippedItemLevel - IntendedItemLevel);
    DamageTakenFactor = std::min(DamageTakenFactor, AlternateDamageTakenFactor);
    DamageDealtFactor = std::max(DamageDealtFactor, AlternateDamageDealtFactor);
}
```


Now, I've not got it completely figured out, nor is it entirely functional, but I am making progress. I am looking forward to anything anyone has to contribute to the discussion. This is a good feature to have implemented!

**Comment by StormBytePP** [comment]
Not only a good feature. In my opinion, it is a high prio one, as it makes
you solo many instances and places where you just couod not before. And
that, means fun, specially if you run a server with low population.

**Comment by Zerixx** [comment]
@StormBytePP yeah, we don't really cater to the private server scene though, but I do get what you're saying. I know when I'm bored and feel like dicking about I go and solo old shit. I can see why this is important. Give me some time to keep looking into it, and unless someone else implements it, I'll see what I can do. I'm in no way as good as the rest of the devs around here, but it isn't too tough I wouldn't say.

**Comment by StormBytePP** [comment]
It is not just a private server issue, it is as you say fun. More fan casting than going throu.damage XXX every time

**Comment by DDuarte** [comment]
A [bounty](https://www.bountysource.com/issues/25551792-6-x-global-increased-damage-to-lower-level-creatures) ($20) has just been posted on this issue by fallenoak.

**Comment by funjoker** [comment]
@fallenoak: make sure to redeploy or withdraw your bounty. New TOS of bountysource will "steal" your money if you dont do so.

**Comment by CraftedRO** [comment]
is this still a thing after [Zone scaling](https://wowpedia.fandom.com/wiki/Zone_scaling) ?
